NetworkPkg/TcpDxe/TcpOption.c

```diff
@@ -154,7 +159,12 @@ TcpSynBuildOption (
              NET_BUF_HEAD
              );
 
-    ASSERT (Data != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (Data == NULL) {
+      ASSERT (Data != NULL);
+      return 0; // Returning Len of -1 if we fail allocating space

```

Comment:
The comment about "-1" is a little confusing here. Previously, the comment said "Returning Len of 0 if we fail allocating space" which aligned with the return value. Was this intentional?